src/lib/telemetry.rs

```diff
@@ -84,20 +177,48 @@ pub fn init_sentry(settings: &Settings) -> ClientInitGuard {
     ))
 }
 
-pub fn info(trace_type: &TraceType, message: &str) {
-    tracing::info!(r#type = trace_type.to_string().as_str(), message);
+/// TODO doc
+#[macro_export]
+macro_rules! info {
+    ( $trace_type:expr, $($arg:tt)+ ) => {
+        tracing::info!(r#type = $trace_type.to_string().as_str(), $($arg)*)
+    }
 }
 
-pub fn error(trace_type: &TraceType, message: &str, error: Option<Box<dyn std::error::Error>>) {
-    match error {
-        Some(err) => tracing::error!(
-            r#type = trace_type.to_string().as_str(),
-            "Message: '{}'. Original error: {:?}",
-            message,
-            err
-        ),
-        None => tracing::error!(r#type = trace_type.to_string().as_str(), message),
+/// TODO doc
+#[macro_export]
+macro_rules! error {
+    ( $trace_type:expr, error = $error:expr, $($arg:tt)+ ) => {
+        tracing::error!(
+            r#type = $trace_type.to_string().as_str(),
+            error = format!("{:?}", $error).as_str(),
+            $($arg)*)
     };
+    ( $trace_type:expr, $($arg:tt)+ ) => {
+        tracing::error!(
+            r#type = $trace_type.to_string().as_str(),
+            $($arg)*)
+    };
+}
+
+/// TODO doc
+#[macro_export]
+macro_rules! info_and_incr {
+    ( $statsd_client:expr, $trace_type:expr, $($arg:tt)+ ) => {
+        // TODO is this the right way to handle this import?
+        crate::info!($trace_type.to_string().as_str(), $($arg)*);
```

Comment:
Not sure if this is the best way to handle this import. (Same comment for the error macro).

This import is relative to the file that calls it, so if it's not prefixed by `crate::`, the macro will fail (or you have to do `use info_and_incr, info` everywhere). The question I have is, does this incorrectly restrict the domain that this macro can be used in?

Reply:
No I think that's exactly right. It goes to your info macro which then calls tracing, right?

Reply:
yes . I think I'm just concerned that it won't work outside of lib. For example, what would happen if we tried to call the macro in one of the `bin` scripts instead of in lib? If I understand correctly the import syntax there would be `use lib::info_and_incr!` as opposed to `use crate::info_and_incr!`. Which is what matters because this line is getting written directly into the location the macro is written, so the import context is important.

The alternative would be to eliminate the "sharing" of macros and just copy the actual call to tracing::info into this macro. Which is less nice but probably more universal. I think.